### DISCUSSION TITLE
actions/upload-artifact@v3

### DISCUSSION THREAD

**Discussion by gl72** [community/community#43954] [Enterprise] [answered]
### Select Topic Area

Question

### Body

Hi all,
I have a workflow with 2 jobs, one is a "preflight" check and the second is the actual deployment.
During the preflight check I upload some json files as artifacts that contain configuration files that will be used during the deployment.
The second job only runs after a repository member approves the workflow run, theoretically after having checked the validity of the json files.
The problem is that the artifacts are available only after the complete workflow execution. 
Would it be possible to make them available only after the preflight job?
Thanks in advance for your reply!
Giancarlo

**Comment by jsoref** [comment, marked as answer]
It'd be nice.

In the interim, you can use a workflow job that triggers other jobs and then depends on them.

**Comment by gl72** [comment]
Hi @jsoref thanks for your hint. Actually the workflow is triggered by a workflow_dispatch webhook event with a number of inputs that are passed through a REST API call, I will implement something like this https://docs.github.com/en/enterprise-server@3.6/rest/actions/workflows#create-a-workflow-dispatch-event to trigger the actual deployment job from the "preflight" job.

**Comment by spenserblack** [comment]
>Would it be possible to make them available only after the preflight job?

I'm tempted to say to use `jobs.<job_id>.needs`, but since it sounds like they're different triggers (and deployment is triggered manually) it sounds like it wouldn't work.

What you could maybe do is, in the second job, hash the artifacts and try to commit that hash in a file (e.g. `hash.txt` or `checksum.txt`). If the hash is the same, committing fails, so the deployment job would fail. This would at least enforce that the contents of the artifacts *must* change for the deployment job to complete.

**Comment by jsoref** [comment]
I'm assuming that `actions/download-artifact@v3` doesn't do what you want -- it has magical access to artifacts before they're generally available...

**Comment by gl72** [comment]
I have achieved the goal creating 2 workflows, the preflight one does his job and once it is completed generates the artifacts I need and then it triggers the second one (actual deployment) using the workflow dispatch event REST API call.
Thanks you all for your help!

**Reply by soumyarpatra** [reply]
I have created a jar and published in github artifact but from another application when I mentioned the details in pom.xml , I am unable to download that jar. Please share the url for github artifact Or share the settings.xml file .

Error :
Error:  Failed to execute goal on project **surcharge-listener**: Could not resolve dependencies for project net.microservices:surcharge-listener:jar:0.0.1: Could not find artifact net.microservices:util-core:jar:1.0.97 in github (https://github.com/Global/*)
